anta/models.py

```diff
@@ -88,34 +104,28 @@ def render(self, **params: str | int | bool) -> AntaCommand:
 
         Returns
         -------
-            command: The rendered AntaCommand.
-                     This AntaCommand instance have a template attribute that references this
-                     AntaTemplate instance.
+            The rendered AntaCommand.
+            This AntaCommand instance have a template attribute that references this
+            AntaTemplate instance.
 
+        Raises
+        ------
+            AntaTemplateRenderError
+                If a parameter is missing to render the AntaTemplate instance.
         """
-        # Create params schema on the fly
-        field_names = [fname for _, fname, _, _ in Formatter().parse(self.template) if fname]
-        # Extracting the type from the params based on the expected field_names from the template
-        fields: dict[str, Any] = {key: (type(params.get(key)), ...) for key in field_names}
-        # Accepting ParamsSchema as non lowercase variable
-        ParamsSchema = create_model(  # noqa: N806
-            "ParamsSchema",
-            __base__=AntaParamsBaseModel,
-            **fields,
-        )
-
         try:
-            return AntaCommand(
-                command=self.template.format(**params),
-                ofmt=self.ofmt,
-                version=self.version,
-                revision=self.revision,
-                template=self,
-                params=ParamsSchema(**params),
-                use_cache=self.use_cache,
-            )
-        except KeyError as e:
+            command = self.template.format(**params)
+        except (KeyError, SyntaxError) as e:
             raise AntaTemplateRenderError(self, e.args[0]) from e
+        return AntaCommand(
+            command=command,
+            ofmt=self.ofmt,
+            version=self.version,
+            revision=self.revision,
+            template=self,
+            params=self.params_schema(**params),
```

Comment:
this could raise with extra Key - don't we want the AntaTemplateRenderError in that case?

Reply:
hmm I see the test is looking at the pydantic error. This is actually an error for test developers so probably good as it is. Let's leave it as it.